### PR TITLE
fix(sql): fix UnsupportedOperationException thrown on sample by with invalid values

### LINKED ISSUE #2532: UnsupportedOperationException on syntax error with sample by
### Describe the bug

This works fine: 
`select timestamp, avg(price) from orders sample by 1h align to calendar with OFFSET '00:00';`
I accidentally executed the query as:
`select timestamp, avg(price) from orders sample by 1h align to calendar with OFFSET 0;`
and this logged this exception on a server side:
```
2022-09-19T11:05:32.711166Z I http-server connected [ip=172.18.0.1, fd=8]
2022-09-19T11:05:32.712531Z I i.q.c.h.p.JsonQueryProcessorState [8] exec [q='select timestamp, avg(price) from orders sample by 1h align to calendar with OFFSET 0;
']
2022-09-19T11:05:32.712601Z I i.q.c.h.p.QueryCache hit [thread=questdb-worker-5, sql=select timestamp, avg(price) from orders sample by 1h align to calendar with OFFSET 0;
]
2022-09-19T11:05:32.712641Z I i.q.c.h.p.JsonQueryProcessorState [8] execute-cached [skip: 0, stop: 1000]
2022-09-19T11:05:32.716545Z E i.q.c.h.p.JsonQueryProcessorState [8] Uh-oh. Error!
java.lang.UnsupportedOperationException
	at io.questdb.griffin.engine.functions.IntFunction.getStr(IntFunction.java:110)
	at io.questdb.griffin.engine.groupby.AbstractSampleByCursor.parseParams(AbstractSampleByCursor.java:106)
	at io.questdb.griffin.engine.groupby.AbstractNoRecordSampleByCursor.of(AbstractNoRecordSampleByCursor.java:107)
	at io.questdb.griffin.engine.groupby.AbstractSampleByRecordCursorFactory.initFunctionsAndCursor(AbstractSampleByRecordCursorFactory.java:76)
	at io.questdb.griffin.engine.groupby.AbstractSampleByNotKeyedRecordCursorFactory.getCursor(AbstractSampleByNotKeyedRecordCursorFactory.java:51)
	at io.questdb.cutlass.http.processors.JsonQueryProcessorState.of(JsonQueryProcessorState.java:668)
	at io.questdb.cutlass.http.processors.JsonQueryProcessorState.of(JsonQueryProcessorState.java:660)
	at io.questdb.cutlass.http.processors.JsonQueryProcessor.executeSelect(JsonQueryProcessor.java:530)
	at io.questdb.cutlass.http.processors.JsonQueryProcessor.executeCachedSelect(JsonQueryProcessor.java:413)
	at io.questdb.cutlass.http.processors.JsonQueryProcessor.execute0(JsonQueryProcessor.java:173)
	at io.questdb.cutlass.http.processors.JsonQueryProcessor.onRequestComplete(JsonQueryProcessor.java:231)
	at io.questdb.cutlass.http.HttpConnectionContext.handleClientRecv(HttpConnectionContext.java:575)
	at io.questdb.cutlass.http.HttpConnectionContext.handleClientOperation(HttpConnectionContext.java:253)
	at io.questdb.cutlass.http.HttpServer$1.lambda$$0(HttpServer.java:92)
	at io.questdb.network.AbstractIODispatcher.processIOQueue(AbstractIODispatcher.java:179)
	at io.questdb.cutlass.http.HttpServer$1.run(HttpServer.java:104)
	at io.questdb.mp.Worker.run(Worker.java:118)

2022-09-19T11:05:32.716759Z I i.q.c.h.HttpConnectionContext kicked out [fd=8]
2022-09-19T11:05:32.716773Z I http-server scheduling disconnect [fd=8, reason=5]
2022-09-19T11:05:32.716820Z I http-server disconnected [ip=172.18.0.1, fd=8, src=queue]
2022-09-19T11:05:32.717382Z I i.q.c.h.p.QueryCache push [thread=questdb-worker-8, sql=select timestamp, avg(price) from orders sample by 1h align to calendar with OFFSET 0;
```
The client was kicked out hence it showed a very unfriendly error and the list of tables disappeared:
image

This would be very hard to troubleshoot, especially in a cloud env. with a limited access to server logs. 

### To reproduce

_No response_

### Expected Behavior

I would expect server to return a proper error message informing user the syntax for `with offset` with wrong. 

### Environment

```markdown
- **QuestDB version**: 6.5.2
```

### Additional context

this is via webconsole, I haven't tried pgwire, but I would expect the same problem.

### PR DESCRIPTION
**Pull request by puzpuzpuz**
Fixes #2532

### CODE REVIEW

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 8 / 8 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/AbstractSampleByCursor.java|8|8|100.00%|

File: core/src/main/java/io/questdb/griffin/engine/groupby/AbstractSampleByCursor.java [hunk 1]
```diff
@@ -103,7 +108,12 @@ protected void parseParams(RecordCursor base, SqlExecutionContext executionConte
             this.nextDstUTC = Long.MAX_VALUE;
         }
 
-        final CharSequence offset = offsetFunc.getStr(null);
+        final CharSequence offset;
+        try {
+            offset = offsetFunc.getStr(null);
+        } catch (UnsupportedOperationException e) {
```
Comment (bluestreak01) [inline core/src/main/java/io/questdb/griffin/engine/groupby/AbstractSampleByCursor.java:114]: `UnsupportedOperationException` is meant to be a placeholder to highlight code inconsistencies in development process. It never meant to be a runtime check.

Perhaps we need to do check upstream somewhere that function is of correct type?

**Review by bluestreak01 — CHANGES_REQUESTED** [review]

File: core/src/main/java/io/questdb/griffin/engine/groupby/AbstractSampleByCursor.java (hunk 1 above)
Comment (puzpuzpuz) [reply core/src/main/java/io/questdb/griffin/engine/groupby/AbstractSampleByCursor.java]: Fair point. Addressed in b2928396e75ba80c8cab72bed5358edd86eb1607

**Review by bluestreak01 — APPROVED** [review]

**Comment by puzpuzpuz** [comment]
Pushed a fix to consider bind variables for both offset and timezone: https://github.com/questdb/questdb/pull/2540/commits/96f1d1a769c960ce7648032417420bb494df4ee4

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 11 / 11 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/WhereClauseParser.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/SqlCodeGenerator.java|10|10|100.00%|

**Review by jerrinot — APPROVED** [review]

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 11 / 11 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/WhereClauseParser.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/SqlCodeGenerator.java|10|10|100.00%|

**Pull request merged**